src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -110,12 +118,62 @@ protected Object readResolve() throws ObjectStreamException {
      * It will also be false for something that has finished but is pending child node creation,
      * such as a completed fork branch which is waiting for the join node to be created.
      * <p>This can only go from true to false and is a shortcut for {@link FlowExecution#isCurrentHead}.
+     * @see #isActive
      */
     @Exported
     public final boolean isRunning() {
         return getExecution().isCurrentHead(this);
     }
 
+    private static final Map<FlowExecution, Set<BlockStartNode>> unclosedBlocks = new WeakHashMap<>();
```

Comment:
and by 'information on running flows' I mean that this cache won't have the current unclosed blocks present I think?